dbconn/reload_credentials.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 Block, Inc.
+
+package dbconn
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	dsndriver "github.com/go-mysql/hotswap-dsn-driver"
+	"github.com/go-sql-driver/mysql"
+
+	"github.com/cashapp/blip"
+	"github.com/cashapp/blip/event"
+)
+
+func init() {
+	dsndriver.SetHotswapFunc(Repo.ReloadDSN)
+}
+
+type Credentials struct {
+	Hostname string
+	Username string
+	Password string
+	TLS      *blip.ConfigTLS
+}
+
+type CredentialFunc func(context.Context) (*Credentials, error)
```

Comment:
https://hackmysql.com/golang/go-antipatterns/#writable-read-only-value

`Credentials` are definitely not something we want code to accidentally/inadvertently modify because the same pointer is passed around.